### ISSUE TITLE
Proxy failed to start with exit code -6

### ISSUE THREAD

**Issue by ggravlingen** [jupyterhub/jupyterhub#4345]
### Bug description

I'm trying to upgrade to 3.1.1 from 2.3.0, but am meeting the errors from the log below. Can't really understand why this is happening and after searching through the issues here, I can't seem to find anyone else who's seen the same issues.

#### Expected behaviour

Jupyterhub starts and I'm able to access the hub.

#### Actual behaviour

The container crashes and I'm unable to access the hub.

### Your personal set up

 - OS: Running this in a Docker environment on Ubuntu 20.04.1.
 
 - Version(s): `jupyterhub/jupyterhub:3.1.1`
 

<details><summary>Full environment</summary>

```
# paste output of `pip freeze` or `conda list` here
```
</details>

<details><summary>Configuration</summary>

```python
import os
import shutil
from pathlib import Path

from dockerspawner import DockerSpawner
from oauthenticator.azuread import LocalAzureAdOAuthenticator
from traitlets import default

c = get_config()


# Spawn a new docker for each user
NOTEBOOK_DIR = "/home/jovyan/work"
HOST_HOME_PATH = os.getenv("HOST_HOME_PATH", "/jupyter-notebook")
HOST_USER_PATH = HOST_HOME_PATH + "/{username}"
HOST_NOTEBOOK_PATH = f"{HOST_USER_PATH}/work/"


def create_folder_if_not_exists(volume: str) -> None:
    """Util function to create a folder."""
    if not os.path.exists(volume):
        path = Path(volume)
        path.mkdir(parents=True, exist_ok=True)
        os.chmod(path, 0o755)


class NewDockerSpawner(DockerSpawner):
    # Users that may access analytical shared folder
    team: list = [
        "ggravlingen",
    ]

    # A folder that can be accessed by all users
    shared_all = f"{HOST_HOME_PATH}/shared-all"

    # Permission based folders
    shared_team_folder = f"{HOST_HOME_PATH}/shared-analytical"

    def start(self):
        """Handle shared volumes."""
        create_folder_if_not_exists(self.shared_team_folder)

        # A folder that can be accessed by all users
        self.volumes[self. shared_team_folder] = {
            "bind": "/home/jovyan/work/0-shared-all",
            "mode": "rw",
        }
        self.volumes[HOST_NOTEBOOK_PATH] = {"bind": NOTEBOOK_DIR, "mode": "rw"}

        return super().start()


class AzureAdOAuthenticator(LocalAzureAdOAuthenticator):
    """Custom implementation that uses username instead of name as login."""

    @property
    def oauth_callback_url(self) -> str:
        """Return callback url."""
        # For local debugging
        return "http://localhost/datalab/hub/oauth_callback"

    @default("username_claim")
    def _username_claim_default(self):
        return "unique_name"

    def normalize_username(self, username):
        """
        Override normalize_username to avoid lowercasing usernames.
        """
        return username.split("@")[0]


def pre_spawn_hook(spawner):
    """
    Pre-spawn hook.

    These commands are run on the host.
    """
    spawner.log.info(f"----------- Running pre-spawn hook -------------------")
    username = spawner.user.name  # get the username
    volume = f"{HOST_HOME_PATH}/{username}/work"

    # path on the jupytherhub host, create a folder based on username if not exists
    create_folder_if_not_exists(volume)
    shutil.chown(volume, user=username, group="users")


c.Application.log_level = "DEBUG"
# The date format used by logging formatters for %(asctime)s
c.Application.log_datefmt = "%Y-%m-%d %H:%M:%S"


c.JupyterHub.spawner_class = NewDockerSpawner
network_name = os.environ["DOCKER_NETWORK_NAME"]
c.DockerSpawner.use_internal_ip = True
c.DockerSpawner.image = os.environ["DOCKER_JUPYTER_IMAGE"]
c.DockerSpawner.network_name = network_name
c.DockerSpawner.notebook_dir = NOTEBOOK_DIR
c.DockerSpawner.remove = True
c.DockerSpawner.extra_create_kwargs = {"user": "root"}
c.DockerSpawner.extra_host_config = {"network_mode": network_name}
c.DockerSpawner.env_keep = [
    "DB_HOST",
]

c.DockerSpawner.environment = {
    "CHOWN_HOME": "yes",
    "CHOWN_EXTRA": "/home/jovyan",
    "CHOWN_HOME_OPTS": "-R",
    "NB_UID": 1000,
    "NB_GID": 1000,
}

c.Spawner.pre_spawn_hook = pre_spawn_hook

# Test config
c.Spawner.default_url = "/lab"

# Don't stop clients when restarting the bub
c.JupyterHub.cleanup_servers = False

c.JupyterHub.services = [
    {
        "name": "cull_idle",
        "admin": True,
        "command": ["python3", "/srv/jupyterhub/cull_idle_servers.py", "--timeout=600"],
    }
]

# Login using AAD
c.JupyterHub.authenticator_class = AzureAdOAuthenticator
c.AzureAdOAuthenticator.oauth_callback_url = "http://localhost/hub/oauth_callback"

c.AzureAdOAuthenticator.tenant_id = os.environ.get("AAD_TENANT_ID")
c.AzureAdOAuthenticator.client_id = os.environ.get("OAUTH_CLIENT_ID")
c.AzureAdOAuthenticator.client_secret = os.environ.get("OAUTH_CLIENT_SECRET")

# specify users and admin
c.Authenticator.admin_users = {"sg06"}
c.AzureAdOAuthenticator.create_system_users = True

# Proxy settings
c.ConfigurableHTTPProxy.debug = False
c.ConfigurableHTTPProxy.log_level = "info"
c.ConfigurableHTTPProxy.pid_file = "/run/jupyterhub-proxy.pid"

# Debug - disable proxy
# start manually
# configurable-http-proxy --ip '' --port 8000 --api-ip 127.0.0.1 --api-port 8001 -error-target http://hub:8081/datalab/hub/error --log-level ‘info’
# c.ConfigurableHTTPProxy.should_start = False
# c.ConfigurableHTTPProxy.auth_token = "CONFIGPROXY_AUTH_TOKEN" 

# IP configuration for the proxy
c.JupyterHub.ip = ""
c.JupyterHub.hub_connect_ip = "hub"
c.JupyterHub.base_url = "/datalab"
```
</details>

<details>

```bash
[I 2023-02-10 08:11:48.136 JupyterHub app:2879] Initialized 0 spawners in 0.004 seconds
[I 2023-02-10 08:11:48.144 JupyterHub metrics:278] Found 0 active users in the last ActiveUserPeriods.twenty_four_hours
[I 2023-02-10 08:11:48.145 JupyterHub metrics:278] Found 0 active users in the last ActiveUserPeriods.seven_days
[I 2023-02-10 08:11:48.146 JupyterHub metrics:278] Found 0 active users in the last ActiveUserPeriods.thirty_days
[W 2023-02-10 08:11:48.146 JupyterHub proxy:625] Found proxy pid file: /run/jupyterhub-proxy.pid
[W 2023-02-10 08:11:48.147 JupyterHub proxy:637] Proxy no longer running at pid=9
[D 2023-02-10 08:11:48.147 JupyterHub proxy:678] Removing proxy pid file /run/jupyterhub-proxy.pid
[W 2023-02-10 08:11:48.147 JupyterHub proxy:746] Running JupyterHub without SSL.  I hope there is SSL termination happening somewhere else...
[I 2023-02-10 08:11:48.147 JupyterHub proxy:750] Starting proxy @ http://0.0.0.0:8000/datalab/
[D 2023-02-10 08:11:48.147 JupyterHub proxy:751] Proxy cmd: ['configurable-http-proxy', '--ip', '0.0.0.0', '--port', '8000', '--api-ip', '127.0.0.1', '--api-port', '8001', '--error-target', 'http://hub:8081/datalab/hub/error', '--log-level', 'info']
[D 2023-02-10 08:11:48.148 JupyterHub proxy:670] Writing proxy pid file: /run/jupyterhub-proxy.pid
node[10]: ../src/node_platform.cc:61:std::unique_ptr<long unsigned int> node::WorkerThreadsTaskRunner::DelayedTaskScheduler::Start(): Assertion `(0) == (uv_thread_create(t.get(), start_thread, this))' failed.
 1: 0x7f909adf03cc node::Abort() [/lib/x86_64-linux-gnu/libnode.so.72]
 2: 0x7f909adf045b  [/lib/x86_64-linux-gnu/libnode.so.72]
 3: 0x7f909ae72de2 node::WorkerThreadsTaskRunner::WorkerThreadsTaskRunner(int) [/lib/x86_64-linux-gnu/libnode.so.72]
 4: 0x7f909ae72f16 node::NodePlatform::NodePlatform(int, v8::TracingController*) [/lib/x86_64-linux-gnu/libnode.so.72]
 5: 0x7f909adbb2c8 node::InitializeOncePerProcess(int, char**) [/lib/x86_64-linux-gnu/libnode.so.72]
 6: 0x7f909adbb4ac node::Start(int, char**) [/lib/x86_64-linux-gnu/libnode.so.72]
 7: 0x7f909a329d90  [/lib/x86_64-linux-gnu/libc.so.6]
 8: 0x7f909a329e40 __libc_start_main [/lib/x86_64-linux-gnu/libc.so.6]
 9: 0x565176eb00f5 _start [node]
[C 2023-02-10 08:11:49.217 JupyterHub app:3091] Failed to start proxy
    Traceback (most recent call last):
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/app.py", line 3089, in start
        await self.proxy.start()
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/proxy.py", line 778, in start
        _check_process()
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/proxy.py", line 774, in _check_process
        raise e from None
    RuntimeError: Proxy failed to start with exit code -6
    
[D 2023-02-10 08:11:49.219 JupyterHub application:1026] Exiting application: jupyterhub
```

</details>

**Comment by minrk** [comment]
That looks like a pretty fundamentally broken installation of nodejs. Maybe reinstalling or upgrading node would fix it?

**Comment by ggravlingen** [comment]
@minrk I'm using the official Docker image so I'm thinking more people would have issues in that case. But for the sake of argument, you mean upgrading node as a part of building the Docker image?

**Comment by ggravlingen** [comment]
Oh, the reason was I was running an old version of Docker. I was advised to upgrade when debugging errors in another Docker container that was also causing issues. After having done that, the hub container started without a blink.

**Comment by minrk** [comment]
Ah, good to know! For others' reference, do you still have the version number that didn't work and the version you upgraded to?

**Comment by ggravlingen** [comment]
I am now on Docker version 23.0.1 and I believe I was at Docker version 20.10.4 or similar.

**Comment by aduil** [comment]
I met the same issue, it got solved when I upgrade Docker from version 20.10.9 to 20.10.21
